spec/mongoid/config_spec.rb

```diff
@@ -545,4 +545,34 @@ def self.logger
       expect(Mongoid::Config.log_level).to eq(2)
     end
   end
+
+  context "with an overridden database" do
+    let(:database) do
+      "test_purge_#{Time.now.to_i}"
+    end
+
+    before do
+      Mongoid.override_database(database)
+    end
+
+    after do
+      Mongoid.override_database(nil)
+    end
+
+    describe "#purge!" do
+      it 'respects persistence context overrides' do
+        House.create!(name: '1', model: 'Big')
+        Mongoid.purge!
+        expect(House.count).to be == 0
```

Comment:
For symmetry can this test please assert the count is 1 before purging/truncation, and I believe the equality can be expressed as

```
        expect(House.count).to eq 0
```